### DISCUSSION TITLE
How to keep `SelectableLabel` from wrapping?

### DISCUSSION THREAD

**Discussion by Barugon** [emilk/egui#2882] [Q&A]
Does anyone know how I can keep `SelectableLabel` from wrapping?

**Comment by j-n-f** [comment]
`Label` has it (https://docs.rs/egui/latest/egui/widgets/struct.Label.html#method.truncate), but not `SelectableLabel`. I thought about copying the logic over, but the widgets are different enough that it makes it difficult.

**Comment by YgorSouza** [comment]
If you set `ui.style_mut().wrap = Some(false);`, the widget will just overflow instead of wrapping or truncating. If you want it to truncate like the label, I think you have to make your own `LayoutJob`.

```rs
let text = "Sphinx of black quartz, judge my vow";
let mut job = LayoutJob::default();
let format = TextFormat{
    font_id: TextStyle::Button.resolve(ui.style()),
    ..Default::default()
};
job.append(text, 0.0, format);
job.wrap = TextWrapping {
    max_rows: 1,
    break_anywhere: true,
    ..Default::default()
};
ui.add(Label::new(text));
ui.add(Label::new(text).truncate(true));
ui.selectable_label(true, job);
ui.button(text);
```

![image](https://github.com/emilk/egui/assets/43298013/65f0f0a8-065e-4cef-8099-41d3cd4a1f28)

I don't know if there is an easier way.

**Reply by j-n-f** [reply]
Nice. I was trying to do it via `TextWrapping::max_width` without success. Your approach works.

**Reply by Barugon** [reply]
> If you set `ui.style_mut().wrap = Some(false);`, the widget will just overflow instead of wrapping or truncating.

Thanks.
